### PR TITLE
sqlite sniffer: check table names for equality

### PR DESCRIPTION
**Pull request by bernt-matthias**
Not sure if the derived classes always give the full list of tables. If so a check for equality (instead for subset) this would be better. 

Unfortunately not all have tests.

Otherwise I would add a parameter that can control if the check should be done for equality

### CODE REVIEW

File: lib/galaxy/datatypes/binary.py [hunk 1]
```diff
@@ -1403,10 +1403,7 @@ def sniff_table_names(self, filename, table_names):
             tables_query = "SELECT name FROM sqlite_master WHERE type='table' ORDER BY name"
             result = c.execute(tables_query).fetchall()
             result = [_[0] for _ in result]
-            for table_name in table_names:
-                if table_name not in result:
-                    return False
-            return True
+            return set(result) == set(table_names)
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/binary.py]: You can  also do `result = (_[0] for _ in result)`. And here I'd use `set(result).issubset(set(table_names))` to maintain the logic.

Comment (nsoranzo) [reply lib/galaxy/datatypes/binary.py]: Also `set(result).issubset(table_names)` works.

**Comment by bernt-matthias** [comment]
I restored the subset check as default behavior and added the possibility to check for equality.

This will be needed here https://github.com/galaxyproject/galaxy/pull/8109 for the OSW and PQP datatypes since the tables of OSW are a superset of the tables of PQP.

**Comment by bernt-matthias** [comment]
I solved it differently ... should we keep the PR .. maybe removing the equality check again but use the issubset function?

File: lib/galaxy/datatypes/binary.py [hunk 2]
```diff
@@ -1395,18 +1395,23 @@ def sniff(self, filename):
...
             conn = sqlite.connect(filename)
             c = conn.cursor()
             tables_query = "SELECT name FROM sqlite_master WHERE type='table' ORDER BY name"
             result = c.execute(tables_query).fetchall()
-            result = [_[0] for _ in result]
-            for table_name in table_names:
-                if table_name not in result:
-                    return False
-            return True
+            result = (_[0] for _ in result)
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/binary.py]:
```suggestion
            result = set(_[0] for _ in result)
```

File: lib/galaxy/datatypes/binary.py [hunk 3]
```diff
@@ -1395,18 +1395,23 @@ def sniff(self, filename):
...
+            if equality:
+                return set(table_names) == set(result)
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/binary.py]:
```suggestion
                return set(table_names) == result
```

File: lib/galaxy/datatypes/binary.py [hunk 4]
```diff
@@ -1395,18 +1395,23 @@ def sniff(self, filename):
...
+            else:
+                return set(table_names).issubset(set(result))
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/binary.py]:
```suggestion
                return set(table_names).issubset(result)
```

**Comment by bernt-matthias** [comment]
I guess we can drop this. Solved it differently.

**Pull request closed without merging**